### ISSUE TITLE
Unitful v0.17.0 breaks tests

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaDiffEq/OrdinaryDiffEq.jl#923]
It seems like `zero(u)` for `u` an array of heterogeneous units fails. 

https://github.com/PainterQubits/Unitful.jl/pull/266
https://github.com/PainterQubits/Unitful.jl/issues/264

I think we should just do some type piracy so it works like a normal array type.

**Comment by mauro3** [comment]
Is there a fix in sight?  Or what's the work-around?

**Comment by ChrisRackauckas** [comment]
I think the work around is to type pirate `zero(u)` to be correct with its types?

**Comment by mauro3** [comment]
But the fix lies with Unitful, right?

**Comment by ChrisRackauckas** [comment]
Yes, I think we need to just open issues and mention that, if Unitful is the only package that makes `zero(u)` return a different type than `u`, it's very difficult to handle generically.

**Comment by mauro3** [comment]
Ok, I'll see whether I can file one.  But it will be a few days until I get the time to look at it.  The easiest work-around is probably to pin Unitful to 0.16.
